buf/validate/internal/message_rules.cc

```diff
@@ -65,7 +68,8 @@ Rules NewMessageRules(
         if (fdesc == nullptr) {
           return absl::FailedPreconditionError(absl::StrCat("field \"", name, "\" not found in message ", descriptor->full_name()));
         }
-        fields.push_back(fdesc);
+        fields.push_back(fdesc); 
+        allMsgOneofs.insert(name)      ; 
```

Comment:
```suggestion
        allMsgOneofs.insert(name);
```
Something odd happened to the formatting on this line.